### PR TITLE
Merge 3.1.11 to master branch

### PR DESCRIPTION
**Pull request by KengoTODA**
Merge all changes in `release-3.1` branch into `master` branch.

I use cherry-pick to avoid revert commit introduced by #850, or we face that `docs/conf.py` and others will lost data made only in `master` branch.

[//]: # (rtdbot-start)

URL of RTD documents:
en: https://spotbugs.readthedocs.io/en/merge-3.1.11-to-master/
ja: https://spotbugs.readthedocs.io/ja/merge-3.1.11-to-master/

[//]: # (rtdbot-end)

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -53,7 +53,7 @@
 
 # General information about the project.
 project = u'spotbugs'
-copyright = u'2016-2017, spotbugs community'
+copyright = u'2016-2018, spotbugs community'
```
Comment (ThrawnCA) [inline docs/conf.py]: This could be 2019 now.

File: docs/introduction.rst [hunk 2]
```diff
@@ -16,3 +16,14 @@ SpotBugs is platform independent, and is known to run on GNU/Linux, Windows, and
...
 To analyze very large projects, more memory may be needed.
+
+Supported Java version
+----------------------
+
+SpotBugs is built by JDK8, and run on JRE8 and newer versions.
+
+SpotBugs can scan bytecode (class files) generated by JDK8 and newer versions.
+However, support for Java 11 and newer is still experimental.
+Visit `issue tracker <https://github.com/spotbugs/spotbugs/issues>`_ to find known problems.
```
Comment (ThrawnCA) [inline docs/introduction.rst]: Missing "the" - should be "visit the issue tracker"

File: eclipsePlugin/doc/building_spotbugs_plugin.txt [hunk 3]
```diff
@@ -80,3 +80,15 @@ Build executable plug-in
 16) Right click on [eclipsePlugin] to refresh the Eclipse workspace
     a) In the [eclipsePlugin]/build folder you will find the site/eclipse-daily folder which contains plugin update site.
     b) Use the path to this folder in [Help | Install New Software] as the path to the update site to install it into Eclipse.
+
+Sign plug-in (optional)
+=======================
+17) If you want to sign your plug-in to avoid security warning,
```
Comment (ThrawnCA) [inline eclipsePlugin/doc/building_spotbugs_plugin.txt]: "avoid security warnings"

File: spotbugs/src/main/java/edu/umd/cs/findbugs/ba/npe/IsNullValueAnalysis.java [hunk 4]
```diff
@@ -668,7 +668,7 @@ private IsNullConditionDecision getDecision(BasicBlock basicBlock, IsNullValueFr
             }
             short secondToLastOpcode = prev.getInstruction().getOpcode();
             // System.out.println("Second last opcode: " +
-            // Const.OPCODE_NAMES[secondToLastOpcode]);
+            // Const.Const.getOpcodeName(secondToLastOpcode));
```
Comment (ThrawnCA) [inline spotbugs/src/main/java/edu/umd/cs/findbugs/ba/npe/IsNullValueAnalysis.java]: Incorrect string replacement.

File: spotbugs/src/main/java/edu/umd/cs/findbugs/classfile/ClassDescriptor.java [hunk 5]
```diff
@@ -1,6 +1,6 @@
 /*
  * FindBugs - Find Bugs in Java programs
- * Copyright (C) 2006, University of Maryland
+ * Copyright (C) 2006-2018, University of Maryland
```
Comment (ThrawnCA) [inline spotbugs/src/main/java/edu/umd/cs/findbugs/classfile/ClassDescriptor.java]: This isn't maintained by the University of Maryland any more.

File: spotbugs/src/main/java/edu/umd/cs/findbugs/classfile/engine/ClassParserUsingASM.java [hunk 6]
```diff
@@ -156,6 +158,8 @@
 
         HashSet<Label> labelsSeen = new HashSet<>();
 
+        private int parameterCount = -1;
```
Comment (ThrawnCA) [inline spotbugs/src/main/java/edu/umd/cs/findbugs/classfile/engine/ClassParserUsingASM.java]: What does a negative count mean?

File: spotbugs/src/main/java/edu/umd/cs/findbugs/util/ClassName.java [hunk 7]
```diff
@@ -1,6 +1,6 @@
 /*
  * FindBugs - Find Bugs in Java programs
- * Copyright (C) 2006, University of Maryland
+ * Copyright (C) 2006-2018, University of Maryland
```
Comment (ThrawnCA) [inline spotbugs/src/main/java/edu/umd/cs/findbugs/util/ClassName.java]: This isn't maintained by UoM anymore.

**Comment by KengoTODA** [comment]
@ThrawnCA all changes have been merged to `release-3.1` and this PR is just applying them to `master`. If we need to fix them, we need to fix in `release-3.1` and release as 3.1.12. Then we can apply them into `master` in the next synchronization.

So firstly could we merge this PR?

**Pull request merged**